src/stack.rs

```diff
@@ -1,15 +1,55 @@
-use crate::bitset::Bitset;
-use std::alloc::{alloc, alloc_zeroed, Layout};
+use crate::{bitset::Bitset, fat_pointer::FatPointer, hash_for_debugging};
+use std::{
+    alloc::{alloc, alloc_zeroed, Layout},
+    fmt,
+};
 use u256::U256;
 
-#[derive(PartialEq, Debug)]
+/// Helper wrapper for debugging [`Stack`] / [`StackSnapshot`] contents.
+struct StackStart<I>(I);
+
+impl<I: Iterator<Item = (bool, U256)> + Clone> fmt::Debug for StackStart<I> {
```

Comment:
I'd put all the Debug related code at the bottom of the file, as it is least relevant for how the code works.